proxy/http2/test_HPACK.cc

```diff
@@ -0,0 +1,413 @@
+  if (compare_header_fields(&decoded, &original) != 0) {
+    result = seqnum;
+    return result;
+  }
+  pack(encoded, written, packed);
+  ofs << "      \"wire\": \"" << packed << "\"" << endl;
+  ofs << "    }" << endl;
+  decoded.destroy();
+  original.destroy();
+
+  ofs << "  ]," << endl;
+  ofs << "  \"description\": \"Apache Traffic Server\"" << endl;
+  ofs << "}" << endl;
+
+  return result;
+}
+
+int
+prepare()
+{
+  filename_in = input_dir + "story_00.json";
+  filename_out = output_dir + "story_00.json";
+  offset_in = input_dir.length() + 6;
+  offset_out = output_dir.length() + 6;
+
+  struct dirent *d;
+  DIR *dir = opendir(input_dir.c_str());
+
+  if (dir == NULL) {
+    cerr << "Cannot open " << input_dir << endl;
+    return 1;
+  }
+  while ((d = readdir(dir)) != NULL) {
+    if (d->d_type == DT_REG) {
+      ++last;
+    }
+  }
+  closedir(dir);
+  cerr << last << " test stories" << endl;
+
+  if (mkdir(output_dir.c_str(), 0755) < 0 && errno != EEXIST) {
+    cerr << "Cannot create output directory" << endl;
+    return 1;
+  }
+  return 0;
+}
+
+REGRESSION_TEST(HPACK_Decoding)(RegressionTest *t, int /* atype ATS_UNUSED */, int *pstatus)
+{
+  TestBox box(t, pstatus);
+  box = REGRESSION_TEST_PASSED;
+
+  int result = -1;
+
+  for (int i = first; i < last; ++i) {
+    filename_in[offset_in + 0] = '0' + i / 10;
+    filename_in[offset_in + 1] = '0' + i % 10;
+    result = test_decoding(filename_in);
+    box.check(result == -1, "Story %d sequence %d failed.", i, result);
+    if (result != -1) {
+      break;
+    }
+  }
+}
+
+REGRESSION_TEST(HPACK_Encoding)(RegressionTest *t, int /* atype ATS_UNUSED */, int *pstatus)
+{
+  TestBox box(t, pstatus);
+  box = REGRESSION_TEST_PASSED;
+
+  int result = -1;
+
+  for (int i = first; i < last; ++i) {
+    filename_in[offset_in + 0] = '0' + i / 10;
+    filename_in[offset_in + 1] = '0' + i % 10;
+    filename_out[offset_out + 0] = '0' + i / 10;
+    filename_out[offset_out + 1] = '0' + i % 10;
+    result = test_encoding(filename_in, filename_out);
+    box.check(result == -1, "Story %d sequence %d failed.", i, result);
+    if (result != -1) {
+      break;
+    }
+  }
+}
+
+int
+main(int argc, char **argv)
+{
+  while (argc >= 2) {
+    // -f option disables freelist
+    if (argc >= 2 && argv[1][0] == '-' && argv[1][1] == 'f') {
+      ink_freelist_init_ops(ink_freelist_malloc_ops());
+      --argc;
+      ++argv;
+    }
+    // -i option specifies input dir
+    if (argc >= 3 && argv[1][0] == '-' && argv[1][1] == 'i') {
+      input_dir = argv[2];
+      if (*input_dir.end() != '/') {
+        input_dir += '/';
+      }
+      argc -= 2;
+      argv += 2;
+    }
+    // -o option specifies input dir
+    if (argc >= 3 && argv[1][0] == '-' && argv[1][1] == 'o') {
+      output_dir = argv[2];
+      if (*output_dir.end() != '/') {
+        output_dir += '/';
+      }
+      argc -= 2;
+      argv += 2;
```

Comment:
It would be cleaner to use ``ink_args`` here. I think that it should be OK to pass these extra options down to ``RegressionTest::main``.